fusion-core/src/__tests__/enhance.js

```diff
@@ -166,6 +166,94 @@ tape('enhancement with a plugin with deps', t => {
   app.resolve();
 });
 
+// TODO(#573) This is a regression test which hasn't been fixed yet
```

Comment:
Thoughts on an additional plugin enhancer test where the enhancer's returned plugin has unregistered dependencies?  Something like:

```js
let app = new App('el', el => el);
type FnType = string => string;
const FnToken: Token<FnType> = createToken('FnType');
const EnhancerDepToken: Token<FnType> = createToken('EnhancerDepToken');
let BaseFnEnhancer = (fn: FnType): FusionPlugin<void, FnType> => {
  return createPlugin({
    deps: {
      addlFn: EnhancerDepToken,
    },
    provides: () => {
      return arg => {
        return fn(arg) + ' enhanced, and also ' + addlFn(arg);
      };
    },
  });
};
app.register(FnToken, msg => msg);
app.enhance(FnToken, BaseFnEnhancer);
t.throws(() => {
  app.resolve();
});
t.end();
```

(If we don't already cover this case)

Reply:
We do have this test case